components/edit-collective/sections/UserSecurity.js

```diff
@@ -194,6 +212,143 @@ class UserTwoFactorAuth extends React.Component {
     }
   }
 
+  async setPassword() {
+    const { password, passwordKey, currentPassword, passwordScore } = this.state;
+
+    if (passwordScore <= 1) {
+      this.setState({
+        passwordError: (
+          <FormattedMessage defaultMessage="Password is too weak. Try to use more characters or use a password manager to generate a strong one." />
+        ),
+      });
+      return;
+    }
+
+    try {
+      this.setState({ passwordLoading: true });
+      await this.props.setPassword({
+        variables: { password, currentPassword },
+      });
+      this.props.addToast({
```

Comment:
We should move the toast at the end, after the `refetch` + `setState`; it will make the flow clearer with slow requests. Currently:

![Peek 2023-01-26 11-44](https://user-images.githubusercontent.com/1556356/214817059-44f854d3-00ca-4b4e-82e8-d51a6666464a.gif)